src/federationparams.h

```diff
@@ -35,14 +40,42 @@ class CFederationParams
     /**
      * Parse aggPubkey in block header.
      */
-    CPubKey ReadAggregatePubkey(const std::vector<unsigned char>& pubkey);
-    const CPubKey& GetAggregatePubkey() const{ return aggregatePubkey; }
+    CPubKey ReadAggregatePubkey(const std::vector<unsigned char>& pubkey, uint height);
+    const CPubKey& GetLatestAggregatePubkey() const { return aggregatePubkey.back(); }
     bool ReadGenesisBlock(std::string genesisHex);
     const CBlock& GenesisBlock() const { return genesis; }
     const std::string& getDataDir() const { return dataDir; }
     const std::vector<SeedSpec6>& FixedSeeds() const { return vFixedSeeds; }
     /** Return the list of hostnames to look up for DNS seeds */
     const std::vector<std::string>& DNSSeeds() const { return vSeeds; }
+    const uint& GetHeightFromAggregatePubkey(std::vector<unsigned char> aggpubkey) const {
+        for (auto& c : aggregatePubkeyHeight)
+            if (c.aggpubkey == CPubKey(aggpubkey.begin(), aggpubkey.end())) {
+                return c.height;
+                break;
+            } else {
+                continue;
+            }
```

Comment:
We need a default case when aggpubkey is not in the list.

Reply:
throw error when aggpubkey not in the list?

Reply:
`uint& CFederationParams::GetHeightFromAggregatePubkey(std::vector<unsigned char> aggpubkey) const
{
    for (auto& c : aggregatePubkeyHeight) {
        if (c.aggpubkey == CPubKey(aggpubkey.begin(), aggpubkey.end()))
            return c.height;
    }
    throw std::runtime_error("GetHeightFromAggregatePubkey: aggpubkey not found");
}`

i change it to throw error so it will not process after it gives error ok?

Reply:
no throw runtime_error will stop Tapyrusd. we should just return a default value. I think for this we can return -1 so that we know its not in the list. but to return we have to change the return type to "int" from "uint"